resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -4,10 +4,14 @@ jruby-puppet: {
     # Puppet server expects to load Puppet from this location
     ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
 
-    # This setting determines where JRuby will look for gems.  It is also
-    # used by the `puppetserver gem` command line tool.
+    # This setting determines where JRuby will install gems.  It is used for loading gems,
+    # and also by the `puppetserver gem` command line tool.
     gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
 
+    # This setting defines the complete "GEM_PATH" for jruby.  If set, it should include
+    # the gem-home directory as well as any other directories that gems can be loaded
+    # from (including the vendored gems directory for gems that ship with puppetserver)
+    gem-path: "/opt/puppetlabs/server/data/puppetserver/vendored-jruby-gems:/opt/puppetlabs/server/data/puppetserver/jruby-gems"
```

Comment:
Yeah, I wasn't  totally sure I wanted to introduce our first use of HOCON interpolation just for this, but I think you're right, it's probably worth it.